### ISSUE TITLE
Poedit icon is missing in about dialog

### ISSUE THREAD

**Issue by xfgusta** [vslavik/poedit#704]
#  Steps to reproduce

Help → About on the toolbar.

# Expected behavior
![Old](https://xfgusta.fedorapeople.org/img/old.png)

# Current behavior
![New](https://xfgusta.fedorapeople.org/img/new.png)

# Additional information

I'm running Fedora 34 (Workstation).

[aalam](https://bodhi.fedoraproject.org/users/aalam) was [able to reproduce](https://bodhi.fedoraproject.org/updates/FEDORA-2021-979faa949d#comment-2087680) the issue. I was also able to reproduce it building from [source](https://github.com/vslavik/poedit/releases/download/v3.0-oss/poedit-3.0.tar.gz).

**Comment by vslavik** [comment]
GTK+ version? wxWidgets version? Installation location? Installed icon files and their locations? Reasons why Fedora/GTK+ is not picking the installed icon? You're compiling from sources and are distro person, surely you have some visibility into it and can provide *some* details...

Relevant commit (self-evidently): 45e6b4032b756c6856ec5ef1615ca23cebc3aa93

The removal of >128px icons and their replacement with SVG matches GNOME guidelines as far as I can tell.

**Comment by c72578** [comment]
- GTK+ version: `3.24.29` (3.24.29-1.fc34)
- wxWidgets version: `3.0.5` (3.0.5.1-4.fc34)
- Installation location: `/usr/bin/`
  Further details of installation locations:
  https://koji.fedoraproject.org/koji/rpminfo?rpmID=26523474
- Installed icon files and their locations:
`/usr/share/icons/hicolor/128x128/apps/net.poedit.Poedit.png`
`/usr/share/icons/hicolor/16x16/apps/net.poedit.Poedit.png`
`/usr/share/icons/hicolor/24x24/apps/net.poedit.Poedit.png`
`/usr/share/icons/hicolor/32x32/apps/net.poedit.Poedit.png`
`/usr/share/icons/hicolor/48x48/apps/net.poedit.Poedit.png`
`/usr/share/icons/hicolor/scalable/apps/net.poedit.Poedit.svg`

**Comment by vslavik** [comment]
> GTK3 version: 3.24.29
> GTK+ version: 1.2.10

GTK+ 1 is certainly not used, but GTK+ 2 might. Which GTK+ version is in `ldd poedit` output?

Also, any terminal output? GTK+ is usually pretty chatty when it fails to find something…

**Comment by xfgusta** [comment]
GTK 3:

```
$ ldd /usr/bin/poedit | grep gtk
	libwx_gtk3u_xrc-3.0.so.0 => /lib64/libwx_gtk3u_xrc-3.0.so.0 (0x00007f500e76b000)
	libwx_gtk3u_webview-3.0.so.0 => /lib64/libwx_gtk3u_webview-3.0.so.0 (0x00007f500e733000)
	libwx_gtk3u_adv-3.0.so.0 => /lib64/libwx_gtk3u_adv-3.0.so.0 (0x00007f500e564000)
	libwx_gtk3u_core-3.0.so.0 => /lib64/libwx_gtk3u_core-3.0.so.0 (0x00007f500df62000)
	libgtkspell3-3.so.0 => /lib64/libgtkspell3-3.so.0 (0x00007f500c83b000)
	libgtk-3.so.0 => /lib64/libgtk-3.so.0 (0x00007f500c08b000)
	libwx_gtk3u_html-3.0.so.0 => /lib64/libwx_gtk3u_html-3.0.so.0 (0x00007f500ba69000)
	libwebkit2gtk-4.0.so.37 => /lib64/libwebkit2gtk-4.0.so.37 (0x00007f500811a000)
	libjavascriptcoregtk-4.0.so.18 => /lib64/libjavascriptcoregtk-4.0.so.18 (0x00007f5003c48000)
```
---

There is only one debug message:

```
Debug: window wxMenuBar(0x56237ff3c920, ) lost focus even though it didn't have it
```

There's nothing useful with the `G_MESSAGES_DEBUG=all` flag as well.

I also ran poedit 3.0 with `GTK_DEBUG=interactive` flag. Here is the GtkAboutDialog and its objects:

![image](https://user-images.githubusercontent.com/66979446/122084295-142c2700-cdd8-11eb-8778-5cc04be72e61.png)

The GtkImage child is using "image-missing" icon (clearly).

--- 

I also ran `strace` to see all syscalls, and there is no syscall using "net.poedit.Poedit.svg " as argument.

![image](https://user-images.githubusercontent.com/66979446/122091171-e72f4280-cdde-11eb-8a90-241c627dca1d.png)

**Comment by super7ramp** [comment]
With the following environment:

- Poedit 3.0.1
- wxWidgets 3.1.4 [^1]
- GTK 3.24.31
- openSUSE Tumbleweed 20220128

<details>
<summary>File list</summary>

```
/usr/bin/poedit
/usr/share/applications/net.poedit.Poedit.desktop
/usr/share/applications/net.poedit.PoeditURI.desktop
/usr/share/doc/packages/poedit
/usr/share/doc/packages/poedit/AUTHORS
/usr/share/doc/packages/poedit/NEWS
/usr/share/icons/hicolor/128x128/apps/net.poedit.Poedit.png
/usr/share/icons/hicolor/16x16/apps/net.poedit.Poedit.png
/usr/share/icons/hicolor/24x24/apps/net.poedit.Poedit.png
/usr/share/icons/hicolor/32x32/apps/net.poedit.Poedit.png
/usr/share/icons/hicolor/48x48/apps/net.poedit.Poedit.png
/usr/share/icons/hicolor/scalable/apps/net.poedit.Poedit.svg
/usr/share/licenses/poedit
/usr/share/licenses/poedit/COPYING
/usr/share/man/man1/poedit.1.gz
/usr/share/metainfo/net.poedit.Poedit.appdata.xml
/usr/share/poedit
/usr/share/poedit/icons
/usr/share/poedit/icons/CrowdinLogoTemplate.png
/usr/share/poedit/icons/DownvoteTemplate.png
/usr/share/poedit/icons/ExtractorsGNUgettext.png
/usr/share/poedit/icons/ItemBookmarkTemplate.png
/usr/share/poedit/icons/ItemCommentTemplate.png
/usr/share/poedit/icons/StatusError.png
/usr/share/poedit/icons/StatusErrorBlack.png
/usr/share/poedit/icons/StatusWarning.png
/usr/share/poedit/icons/StatusWarningBlack.png
/usr/share/poedit/icons/SuggestionErrorTemplate.png
/usr/share/poedit/icons/SuggestionPerfectMatch.png
/usr/share/poedit/icons/SuggestionTMTemplate.png
/usr/share/poedit/icons/document-open.png
/usr/share/poedit/icons/document-save.png
/usr/share/poedit/icons/hicolor
/usr/share/poedit/icons/hicolor/scalable
/usr/share/poedit/icons/hicolor/scalable/actions
/usr/share/poedit/icons/hicolor/scalable/actions/poedit-sync-symbolic.svg
/usr/share/poedit/icons/hicolor/scalable/actions/poedit-update-symbolic.svg
/usr/share/poedit/icons/hicolor/scalable/actions/poedit-validate-symbolic.svg
/usr/share/poedit/icons/hicolor/scalable/actions/sidebar-symbolic.svg
/usr/share/poedit/icons/poedit-status-cat-mid.png
/usr/share/poedit/icons/poedit-status-cat-no.png
/usr/share/poedit/icons/poedit-status-cat-ok.png
/usr/share/poedit/icons/poedit-sync.png
/usr/share/poedit/icons/poedit-update.png
/usr/share/poedit/icons/poedit-validate.png
/usr/share/poedit/icons/sidebar.png
/usr/share/poedit/icons/window-close.png
```
</details>

Manually setting an icon to the `wxAboutDialogInfo` workarounds the issue:

```diff
diff -up poedit-3.0.1/src/edapp.cpp.orig poedit-3.0.1/src/edapp.cpp
--- poedit-3.0.1/src/edapp.cpp.orig	2022-01-29 20:45:04.244649859 +0100
+++ poedit-3.0.1/src/edapp.cpp	2022-01-30 13:44:23.175376893 +0100
@@ -1078,6 +1078,7 @@ void PoeditApp::OnAbout(wxCommandEvent&)
     about.SetCopyright(L"Copyright \u00a9 1999-2021 Václav Slavík");
 #ifdef __WXGTK__ // other ports would show non-native about dlg
     about.SetWebSite("https://poedit.net");
+    about.SetIcon(wxArtProvider::GetIcon("net.poedit.Poedit", wxART_FRAME_ICON, wxSize(128, 128)));
 #endif
 
     wxAboutBox(about);
```

I don't understand the cause of the problem though. Launching Poedit with gdb, I can see that:

- [`wxAboutBox(const wxAboutDialogInfo& info, wxWindow* parent)`](https://github.com/wxWidgets/wxWidgets/blob/v3.1.5/src/gtk/aboutdlg.cpp#L157) retrieves icon from given `wxAboutDialogInfo`
- As no icon is manually set to the `wxAboutDialogInfo`, [`wxAboutDialogInfo::GetIcon()`](https://github.com/wxWidgets/wxWidgets/blob/v3.1.5/src/generic/aboutdlgg.cpp#L81) looks for the top level window icon
- The returned icon is not ok so logo is not set

But as far as I understand, default icon is supposed to be set thanks to Poedit calling [`gtk_window_set_default_icon_name`](https://github.com/vslavik/poedit/blob/v3.0.1-oss/src/edapp.cpp#L343) so I don't get where the issue is.

[^1]: Unfortunately I cannot test with wxWidgets 3.0.5 which is the version reported by other users - openSUSE builds  wxWidgets 3.0.5 against GTK2 only (because of [boo#1183267](https://bugzilla.opensuse.org/show_bug.cgi?id=1183267)) and this build misses `wxWebView` required since Poedit 3.0 (because openSUSE has removed WebkitGTK for GTK2).

**Comment by gusnan** [comment]
I can confirm that @super7ramp's fix above works on wxWidgets 3.0.5 with GTK3. (Tested on Debian unstable).

**Comment by vslavik** [comment]
I am as confused as @super7ramp about why this fix works 🤷‍♂️ But never mind that, committed now as d8375bfedb6f31ba84ee785d962f24a024a312a5 (attributed to the actual author).  

Thanks @super7ramp  for figuring it out!
